velox/functions/prestosql/types/JsonType.h

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/expression/CastExpr.h"
+#include "velox/type/Type.h"
+
+namespace facebook::velox {
+
+/// Custom operator for casts from and to Json type.
+class JsonCastOperator : public exec::CastOperator {
+ public:
+  static const std::shared_ptr<CastOperator>& get() {
+    static const std::shared_ptr<CastOperator> instance{new JsonCastOperator()};
+
+    return instance;
+  }
+
+  bool isSupportedType(const TypePtr& other) const override;
+
+  void castTo(
+      const SelectivityVector& rows,
+      exec::EvalCtx* context,
+      const VectorPtr& input,
+      DecodedVector* inputDecoded,
+      const TypePtr& fromType,
+      VectorPtr* result) override;
+
+  void castFrom(
+      const SelectivityVector& /*rows*/,
+      exec::EvalCtx* /*context*/,
+      const VectorPtr& /*input*/,
+      DecodedVector* /*inputDecoded*/,
+      const TypePtr& /*toType*/,
+      VectorPtr* /*result*/) override {
+    VELOX_NYI();
+  }
+
+ private:
+  JsonCastOperator() = default;
+
+  template <TypeKind kind>
+  void generateJsonTyped(const DecodedVector* inputDecoded, int row);
```

Comment:
this method doesn't need to be in the header file; move it to .cpp; ditto generateJson